packages/rpc-core/src/index.ts

```diff
@@ -117,23 +123,40 @@ export default class Rpc implements RpcInterface {
   private createMethodSend (method: RpcMethod): RpcInterface$Method {
     const rpcName = `${method.section}_${method.method}`;
 
-    const call = async (...values: Array<any>): Promise<any> => {
+    const call = (...values: Array<any>): Observable<any> => {
       // TODO Warn on deprecated methods
-      try {
-        const params = this.formatInputs(method, values);
-        const paramsJson = params.map((param) => param.toJSON());
-        const result = await this._provider.send(rpcName, paramsJson);
-
-        return this.formatOutput(method, params, result);
-      } catch (error) {
-        const message = `${Rpc.signature(method)}:: ${error.message}`;
 
-        l.error(message);
-
-        throw new ExtError(message, (error as ExtError).code, undefined);
-      }
+      // Here, logically, it should be `of(this.formatInputs(method, values))`.
+      // However, formatInputs can throw, and when it does, the above way
+      // doesn't throw in the "Observable loop" (which is internally wrapped in
+      // a try/catch block). So we:
+      // - first do `of(1)` - won't throw
+      // - then do `map(()=>this.formatInputs)` - might throw, but inside Observable.
+      return of(1)
+        .pipe(
+          map(() => this.formatInputs(method, values)),
```

Comment:
just for info `defer(() => of(this.formatInputs(method, values))` also does the trick